dumpling/dumpling.go

```diff
@@ -85,7 +85,6 @@ func (m *Dumpling) Process(ctx context.Context, pr chan pb.ProcessResult) {
 	failpoint.Inject("dumpUnitProcessForever", func() {
 		m.logger.Info("dump unit runs forever", zap.String("failpoint", "dumpUnitProcessForever"))
 		<-ctx.Done()
-		failpoint.Return()
```

Comment:
though its name is "run forever", I guess it should return to skip below logic here, because in this way so `ProcessResult` is sent so imitate "run forever"

Reply:
Dumpling not support context before, so return here. Now we support context.Cancel, below logic will deal with it?